### PR TITLE
Feat have a big number visualisation type

### PR DESCRIPTION
**Pull request by nathaliabj**
### Pre requirements:
- [ ] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: #940 

### Description:
This PR adds a big number as type chart visualisation, this type doesn't have the functionality to download the chart, and allows you to visualise it when only a metric is selected.

### Preview:
Screenshot 2022-01-31 at 09 34 34

### Scope of the changes (select all that apply):
- [X] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD

### CODE REVIEW

**Comment by nathaliabj** [comment]
@owlas ATM the value is hardcoded, working on the real data now,  but in the mean time could you please help me updating the DB to be able to save the `BIG_NUMBER` chart type 🙏

**Comment by owlas** [comment]
@nathaliabj updated with db file!

File: packages/frontend/src/components/SimpleChart/index.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ const flipXFromChartType = (chartType: DBChartTypes) => {
         case DBChartTypes.COLUMN:
         case DBChartTypes.LINE:
         case DBChartTypes.SCATTER:
+        case DBChartTypes.BIG_NUMBER:
```
Comment (owlas) [inline packages/frontend/src/components/SimpleChart/index.tsx:19]: @nathaliabj you shouldn't need this part

File: packages/frontend/src/components/SimpleChart/index.tsx [hunk 2]
```diff
@@ -35,6 +36,8 @@ const echartType = (chartType: DBChartTypes) => {
             return 'bar';
         case DBChartTypes.SCATTER:
             return 'scatter';
+        case DBChartTypes.BIG_NUMBER:
```
Comment (owlas) [inline packages/frontend/src/components/SimpleChart/index.tsx:39]: nor here

File: packages/frontend/src/components/SimpleStatistic/index.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React, { FC } from 'react';
+import {
+    BigNumber,
+    BigNumberContainer,
+    BigNumberLabel,
+    SimpleStatisticsWrapper,
+} from './SimpleStatistics';
+
+interface Props {
+    data: any;
```
Comment (owlas) [inline packages/frontend/src/components/SimpleStatistic/index.tsx:10]: data type should be `Record<string,any>[]`

File: packages/frontend/src/components/SimpleStatistic/index.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+const SimpleStatistic: FC<Props> = ({ data }) => {
+    const getBigNumber = data;
+    console.log(getBigNumber);
+
+    return (
+        <SimpleStatisticsWrapper>
```
Comment (owlas) [inline packages/frontend/src/components/SimpleStatistic/index.tsx:18]: This looks really good!

**Review by owlas — CHANGES_REQUESTED** [review]

File: packages/frontend/src/components/SimpleStatistic/index.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    SimpleStatisticsWrapper,
+} from './SimpleStatistics';
+
+interface Props {
+    data: any;
+}
+
+const SimpleStatistic: FC<Props> = ({ data }) => {
+    const getBigNumber = data;
+    console.log(getBigNumber);
```
Comment (ZeRego) [inline packages/frontend/src/components/SimpleStatistic/index.tsx:15]: forgotten console.log

File: packages/frontend/src/components/SimpleChart/index.tsx (hunk 1 above)
Comment (ZeRego) [reply packages/frontend/src/components/SimpleChart/index.tsx:19]: I suppose we need to exclude this type from the argument so typescript doesn't complain about the switch not handling that type.

File: packages/frontend/src/components/SimpleStatistic/index.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import React, { FC } from 'react';
+import {
+    BigNumber,
+    BigNumberContainer,
+    BigNumberLabel,
+    SimpleStatisticsWrapper,
+} from './SimpleStatistics';
```
Comment (ZeRego) [inline packages/frontend/src/components/SimpleStatistic/index.tsx:7]: Have we stopped using `.styles.tsx` in the name ?

Comment (nathaliabj) [reply packages/frontend/src/components/SimpleStatistic/index.tsx:7]: Good catch! completely missed that! 👀

**Comment by nathaliabj** [comment]
@owlas @TuringLovesDeathMetal Just missing the download option for this type. how would we like this to occur, as an image ?

**Comment by TuringLovesDeathMetal** [comment]
Yep, an image would be great!

**Comment by owlas** [comment]
Do we need a download option for a big number? Seems unnecessary if it's going to take time @nathaliabj !

File: packages/frontend/src/components/SimpleStatistic/index.tsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+interface Props {
+    data: ChartConfig;
+}
+
+const SimpleStatistic: FC<Props> = ({ data }) => {
+    if (!data.plotData) return null;
+    const metricKey = data.series[0];
+    const bigNumberLabel = data.metricOptions[0].label;
+
+    const bigNumber = Math.max(...data.plotData.map((o) => o[metricKey]));
```
Comment (owlas) [inline packages/frontend/src/components/SimpleStatistic/index.tsx:19]: @nathaliabj does this take the biggest value in the column?

@TuringLovesDeathMetal should this be just the first number on the first row in that column?

**Review by owlas — CHANGES_REQUESTED** [review]
Screenshot 2022-02-01 at 09 30 09

There is logic in `useChartConfig` which is out of date because it doesn't allow me to select the big number type when I have just one metric in my table. This should be possible.

**Comment by ZeRego** [comment]
All the chart buttons are disabled when you don’t have at least 1 dimension + 1 metric. But it should be possible to have only 1 metric for the big number type. 
The easiest way to fix this is to not disable the chart type buttons ( but keep the disabled for config/download buttons ) since the <SimpleChart> already handles the empty state.

**Comment by owlas** [comment]
@ZeRego @nathaliabj I think the other problem is that `chartConfig.plotData` will actually be undefined when there is only 1 metric. 

`useChartConfig` was designed to be used by `SimpleChart` perhaps we can have another hook that is designed to only be consumed by `BigNumber` ?

File: packages/frontend/src/components/SimpleStatistic/index.tsx (hunk 7 above)
Comment (nathaliabj) [reply packages/frontend/src/components/SimpleStatistic/index.tsx:19]: Thought it was the biggest number in the metric 🙈  Just updated it!

File: packages/frontend/src/components/LightdashVisualisation/index.tsx [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    chartType,
+    chartConfig,
+    bigNumberData,
+}) => {
+    if (isLoading && !chartConfig) {
+        return <LoadingState />;
+    }
+    return (
+        <>
+            {chartType === 'big_number' ? (
```
Comment (ZeRego) [inline packages/frontend/src/components/LightdashVisualisation/index.tsx:31]:
```suggestion
            {chartType === DBChartTypes.BIG_NUMBER ? (
```

**Review by ZeRego — APPROVED** [review]

**Review by owlas — APPROVED** [review]

**Pull request merged**